pkg/virt-handler/vm.go

```diff
@@ -2542,6 +2542,16 @@ func (d *VirtualMachineController) vmUpdateHelperDefault(origVMI *v1.VirtualMach
 		if err := d.hotplugVolumeMounter.Mount(vmi); err != nil {
 			return err
 		}
+
+		if util.IsNonRootVMI(vmi) {
+			isolationRes, err := d.podIsolationDetector.Detect(vmi)
+			if err != nil {
+				return fmt.Errorf(failedDetectIsolationFmt, err)
+			}
+			if err := changeOwnershipOfHotplugDisks(vmi, isolationRes); err != nil {
```

Comment:
I see that `nonRootSetup` is called a little bit above this place and it would correctly set the ownership too. Would it make sense to change the order of calling `hotplugVolumeMounter.Mount` and `nonRootSetup` to keep `changeOwnershipOfHotplugDisks completely in `prepareStorage`?

Reply:
I am not sure. We call `nonRootSetup` before VMI is running or on the new target node. Here we are in the scope of running vmi.

Reply:
I see what you mean. Why I am asking: setting the ownership is for my taste a little bit too far away from the actual hotplug code. It feels a little like we are shadowing calls on the hotplug mounter. Maybe we can move it closer to the hotplug code. While I don't think it is an issue performance-wise, we may then also be able to only check and set permissions during an actual hoptlug instead of re-checking them over and over.

Reply:
I see what you mean. The `mount` function should be able to do it. I am going to rework it.